src/core/reactor_backend.cc

```diff
@@ -50,6 +52,11 @@ namespace fs = std::filesystem;
 class pollable_fd_state_completion : public kernel_completion {
     promise<> _pr;
 public:
+    void abort() {
+        if (_pr.waiting_task()) {
+            _pr.set_exception(pollable_fd_abort());
+        }
```

Comment:
No, in my case this happens rarely, when job serving data on serial port stops or restarts.

Without it we still get `broken_promise` exception, because completion instance dies without promise fulfiled.